### PR TITLE
Tests UI v4merged

### PR DESCRIPTION
**Pull request by dlaxcess**
add :
- Nft price in Nft details
- Sell button ( Approve & setPrice )
- Burn button ( remove nft from local storage after burn )
- Buy button

Disable Transfer buuton when not effective
remove nft from local storage after transfer

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -55,4 +55,4 @@
   "workbench.layoutControl.enabled": true,
   "workbench.statusBar.visible": true,
   "favorites.sortOrder": "DESC"
-}
```
Comment (zapaz) [inline .vscode/settings.json:58]: file to remove from modification list

File: svelte/stores/nft/nft.ts [hunk 2]
```diff
@@ -63,13 +63,28 @@ const nftRefresh = async (chainId: number, address: string, tokenID: string): Pr
 const nftGetStore = (chainId: number, address: string, tokenID: string): Readable<NftType> =>
   derived(nftListStore, ($nftListStore) => $nftListStore.get(nftGetKey(chainId, address, tokenID)));
 
+// Remove one nft from store & localstorage and its collection from collectionStore & localStorage
```
Comment (zapaz) [inline svelte/stores/nft/nft.ts:66]: why delete collection ? not only nft

File: svelte/stores/nft/nft.ts [hunk 3]
```diff
@@ -63,13 +63,28 @@ const nftRefresh = async (chainId: number, address: string, tokenID: string): Pr
...
+const nftRemoveOne = (chainId: number, address: string, tokenID: string, account: string) => {
+  const collToRemove = collectionStore.getKey(chainId, address);
+  const nftToRemoveKey = nftGetKey(chainId, address, tokenID);
+
+  get(nftListStore).delete(nftToRemoveKey);
```
Comment (zapaz) [inline svelte/stores/nft/nft.ts:71]: to be done via store.update  (like on line 30), something like 

```
 nftListStore.update(($nftListStore: Map<string, NftType>): Map<string, NftType> => {
    const keyToRemove = nftGetKey(chainId, address, tokenID);

    if (typeof localStorage !== "undefined") {
     localStorage.removeItem(keyToRemove);
    }
    
    return $nftListStore.filter( (key) => key != keyToRemove );
  });


```

File: svelte/components/Nft/NftTransfer.svelte [hunk 4]
```diff
@@ -31,6 +39,9 @@
 
       transfered = Boolean(txReceipt.status);
       transfering = false;
+
+      nftStore.nftRemoveOne(chainId, address, tokenID, $metamaskAccount);
```
Comment (zapaz) [inline svelte/components/Nft/NftTransfer.svelte:43]: an NFTRefresh on NFT component should be sufficient ? no need for a remove function ?

File: common/lib/ktransfer.ts [hunk 5]
```diff
@@ -25,15 +25,22 @@ const transferNftResponse = async (
 
   const { contract, supports } = await collectionGetContract(chainId, address, owner.provider);
```
Comment (zapaz) [inline common/lib/ktransfer.ts:26]: should pass owner as last parameter, then no need of connectedContract afterwards

**Review by zapaz — CHANGES_REQUESTED** [review]
here are first remarks

File: svelte/components/Nft/Nft.svelte [hunk 6]
```diff
@@ -47,6 +56,14 @@
   };
 
   $: console.log("Nft", $nft);
+
+  $: chainId && address && account && checkBurnable();
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:60]: either load collection onLoad , either add collection Info needed for nft to nftType

**Review by zapaz — CHANGES_REQUESTED** [review]
should use yield for all transactions , as they all have 2 phases

File: common/lib/kautomarket.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
+import type { OpenNFTsV4 } from "@soltypes/contracts/next/OpenNFTsV4";
+
+import { JsonRpcSigner, TransactionResponse, TransactionReceipt } from "@ethersproject/providers";
+import { Provider } from "@ethersproject/abstract-provider";
+import { BigNumber, constants, ethers } from "ethers";
+import { collectionGetContract } from "@lib/kcollection-get";
+import { getExplorer, explorerUrl } from "./kconfig";
+
+const getNftPrice = async (chainId: number, address: string, tokenID: string, provider: Provider): Promise<string> => {
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:9]: convertion in knft-get-metadata ?

File: common/lib/kautomarket.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  let price = "";
+
+  const { contract, supports } = await collectionGetContract(chainId, address, provider);
+  if (supports.IOpenMarketable) {
+    const openNFTsV4 = contract as OpenNFTsV4;
+    price = ethers.utils.formatEther((await openNFTsV4.callStatic.tokenPrice(BigNumber.from(tokenID))).toString());
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:15]: indeed not

File: common/lib/kautomarket.ts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+  chainId: number,
+  address: string,
+  tokenID: string,
+  provider: Provider
+): Promise<{ receiver: string; royaltyAmount: string } | undefined> => {
+  const { contract, supports } = await collectionGetContract(chainId, address, provider);
+
+  console.log("🚀 ~ file: kautomarket.ts ~ line 28 ~ contract", contract);
+
+  if (!supports.IOpenMarketable) return undefined;
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:31]: key word "address" can be confusing with "address" of collection ?

File: common/lib/kautomarket.ts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+const getDefaultCollPrice = async (chainId: number, address: string, signer: JsonRpcSigner): Promise<string> => {
+  const { contract, supports } = await collectionGetContract(chainId, address, signer.provider);
+
+  if (!supports.IOpenMarketable) return "";
+
+  return ethers.utils.formatEther(await (contract as OpenNFTsV4).callStatic.defaultPrice());
+};
+
+const checkApprouved = async (
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:52]: done

File: common/lib/kautomarket.ts [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+  address: string,
+  tokenID: string,
+  signer: JsonRpcSigner
+): Promise<TransactionResponse | undefined> => {
+  const { contract, supports } = await collectionGetContract(chainId, address, signer.provider);
+  if (!supports.IOpenMarketable) return;
+  const connectedContract = contract.connect(signer) as OpenNFTsV4;
+  const txResp: TransactionResponse | undefined = await connectedContract["approve(address,uint256)"](
+    address,
+    BigNumber.from(tokenID)
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:80]: seems not. conversion is maid py smartcontract ?

File: common/lib/kburn.ts [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+  chainId: number,
+  address: string,
+  tokenID: string,
+  owner: JsonRpcSigner,
+  account: string
+): Promise<TransactionResponse | null> => {
+  // console.log("burnNftResponse", chainId, address, tokenID);
+
+  let txResp: TransactionResponse | null = null;
+  const network = getNetwork(chainId);
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kburn.ts:18]: done

File: common/lib/kburn.ts [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+  if (!(chainId && address && tokenID && network && owner)) return txResp;
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kburn.ts:20]: done

File: common/lib/kbuy.ts [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+  if (!(chainId && address && tokenID && network && buyer)) return txResp;
+
+  const { contract, supports } = await collectionGetContract(chainId, address, buyer.provider);
+  const connectedContract = contract.connect(buyer);
+
+  if (supports.IOpenNFTsV4) {
+    const buyOpenNftV4 = "buy(uint256)";
+    const buyFunction = connectedContract[buyOpenNftV4] as {
+      (tokenID: BigNumber, paymentSent: { value: string }): Promise<TransactionResponse>;
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kbuy.ts:28]: it's working with BigNumber, but don't anderstand why we don't use the function a declared:
buy(
      tokenID: PromiseOrValue<BigNumberish>,
      overrides?: PayableOverrides & { from?: PromiseOrValue<string> }
    ): Promise<ContractTransaction>;

File: svelte/components/Nft/Nft.svelte (hunk 6 above)
Comment (dlaxcess) [reply svelte/components/Nft/Nft.svelte:60]: done in knft-get-metadata

File: .vscode/settings.json (hunk 1 above)
Comment (dlaxcess) [reply .vscode/settings.json:58]: mean add in .gitignore ?

File: common/lib/kautomarket.ts [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+  const [receiver, royaltyAmount] = await (contract as OpenNFTsV4).callStatic.royaltyInfo(
+    BigNumber.from(tokenID),
+    BigNumber.from("10000")
+  );
+
+  console.log("🚀 ~ file: kautomarket.ts ~ line 34 ~ { receiver, royaltyAmount }", receiver);
+  const royaltyAmountEth = ethers.utils.formatEther(royaltyAmount).toString();
+
+  return { receiver, royaltyAmount: royaltyAmountEth };
```
[the remark this answers is not in the document]
Comment (dlaxcess) [reply common/lib/kautomarket.ts:41]: done

File: .vscode/settings.json (hunk 1 above)
Comment (zapaz) [reply .vscode/settings.json:58]: simply do not commit

**Pull request closed without merging**